### PR TITLE
update links to spec pages

### PR DESCRIPTION
**Pull request by aG0aep6G**
This is a companion PR to <https://github.com/D-Programming-Language/dlang.org/pull/896>. Do not pull without that one.

### CODE REVIEW

**Comment by CyberShadow** [comment]
Auto-merge toggled on

File: std/bitmanip.d [hunk 1]
```diff
@@ -300,7 +300,7 @@ The example above creates a tagged reference to an Object in the struct A. This
 as $(D taggedPointer), except the first argument which must be a class type instead of a pointer type.
 
 $(RED Warning: Don't use $(D taggedClassRef) with references to garbage collected objects, as it will result in
-undefined behaviour. See $(LINK http://dlang.org/garbage.html) for details.)
+undefined behaviour. See $(LINK ../spec/garbage.html) for details.)
```
Comment (CyberShadow) [inline std/bitmanip.d]: Oops, this breaks DDox links. This should either be reverted to an absolute URL, or ideally to a macro which is overridden in `std-ddox.ddoc`.

Comment (aG0aep6G) [inline std/bitmanip.d]: Ugh, right. This is a recurring problem with Ddox. See also: <https://github.com/D-Programming-Language/dlang.org/pull/1082#issuecomment-142636522>.

That should be fixed properly, of course. But that's kinda hard, so I think we have two options right now:
1) Change to absolute URLs and accept that they won't work properly in a local context (releases, git forks).
2) Stay with relative URLs and accept adding some more broken links to the /library/ pages.

I'm for 2. /library/ is still a "preview", so I put more weight on /phobos/ pages working offline, too.

Comment (CyberShadow) [inline std/bitmanip.d]: As I mentioned above, we also have the option of defining a macro that will be defined differently for DDox. So, for example, a `$(PHOBOS2SPEC ...)` macro could be defined to be a relative link to `../spec/$1` by default but redifined to an absolute URL for DDox.

Comment (CyberShadow) [inline std/bitmanip.d]: Or even better, define `ROOT_DIR` to be `/` for DDox, or implement it in DDox/dpl-docs to be a relative link to the root directory :)

Comment (aG0aep6G) [inline std/bitmanip.d]: I just wasted another hour trying to get the DDox pages to build. The furthest I got was

    core.exception.RangeError@../../../../.dub/packages/ddox-0.11.5/source/ddox/parsers/jsonparser.d(458): Range violation

I'm giving up once again. Someone else has to take this on.

Comment (CyberShadow) [inline std/bitmanip.d]: How did you try? The Makefile should take care of building everything.

Comment (aG0aep6G) [inline std/bitmanip.d]: > How did you try? The Makefile should take care of building everything.

I finally managed to build the DDox pages. In the end it only took two strategically placed `-L--no-as-needed`s.

> Or even better, define ROOT_DIR to be / for DDox,

Did that:
https://github.com/D-Programming-Language/dlang.org/pull/1159
https://github.com/D-Programming-Language/druntime/pull/1441
https://github.com/D-Programming-Language/phobos/pull/3839

Locally the links are still broken, of course. And this breaks some that happened to work before.

> or implement it in DDox/dpl-docs to be a relative link to the root directory :)

That would be superior, but also significantly more involved.

**Pull request merged**